docs(migrating): update develop guide and overview (#2834)
carbon-design-system/carbon-website: src/pages/migrating/guide/develop.mdx

<PageDescription>

> Step-by-step guide to updating your code from Carbon v10 to v11

Reviewer: ```suggestion
Step-by-step guide to updating your code from Carbon v10 to v11.
```

```

Next, install the `@carbon/react` package:

```bash
npm install @carbon/react
```

Or, with [Yarn](https://yarnpkg.com/):

```bash
yarn add @carbon/react
```

### Step 2: Styles and Dart Sass

If you're importing styles from `carbon-components`, you can now import styles
directly from `@carbon/react` or the `@carbon/react/scss` folder.

Before you're able to bring in these styles, you'll need to make sure your
project is setup to use Dart Sass. Starting in v11, Carbon styles requires Dart
Sass through the `sass` package in order to compile. This change comes from our
migration to Sass Modules in order to improve our compilation times and overall
project structure.

If you don't have this dependency already in your project, you can install it:

```bash
npm install sass
```

Or, with [Yarn](https://yarnpkg.com/):

```bash
yarn add sass
```

Similarly, if you currently use `node-sass` now is a good time to remove that
dependency from your project. In most situations, Dart Sass is a drop-in
replacement for `node-sass` and should require no changes on your end in order
to use it once you install the dependency.

### Step 3: Setup Dart Sass for your project

> One you have Dart Sass installed, it's important that you configure your project

Reviewer 2: ```suggestion
Once you have Dart Sass installed, it's important that you configure your project
```

```scss
// Feature flags
$feature-flags: (
  enable-columns-16: true,
);
// Options
$css--default-type: true;
$css--reset: true;
// Top-level imports
@import 'carbon-components/scss/globals/scss/vars';
@import 'carbon-components/scss/globals/scss/colors';
@import 'carbon-components/scss/globals/scss/theme';
@import 'carbon-components/scss/globals/scss/spacing';
@import 'carbon-components/scss/globals/scss/typography';
@import 'carbon-components/scss/globals/scss/css--reset';
@import 'carbon-components/scss/globals/scss/css--body';
// Component imports
@import 'carbon-components/scss/components/accordion/accordion';
@import 'carbon-components/scss/components/button/button';
```

If you imported the entrypoint from `carbon-components`, you can now do this
directly from `@carbon/styles` without any additional paths by writing:

```scss
@use '@carbon/react';
```

If you were providing any configuration options before you imported Carbon you
can now provide them using the `with` syntax:

```scss
@use '@carbon/react' with ($css--default-type: true, $css--reset: true);
```

If you were using any feature flags in v10, you can safely remove them in v11.

> _Note: you can also use `@import` to bring in Carbon, if you prefer, although

Reviewer 2: On the website we don't normally do notes like this in italic, we normally just use the inline notification component. Its probably fine to leave it as it though.

Author: @aagonzales great point. One thing I noticed was that I wanted to distinguish between "strong suggestions" where we really want to recommend this thing and "passive suggestions" which were kind of throwaway for completeness.

Would you recommend I group both of these under inline notifications so that it's consistent with the rest of the site? Happy to make the change, just wasn't sure what the convention was

Reviewer 2: Yeah I think that distinction is fine. I talked to Lauren about it and we think the format just needs change. The italic was just feeling a little out of place and we think this format could work better

Author: Sounds good, done!

> _Note: you can also use `@import` to bring in Carbon, if you prefer, although
> `@use` is recommended._

Reviewer 2: ```suggestion
**Note:** you can also use `@import` to bring in Carbon, if you prefer, although
`@use` is recommended.
```

Author: Just went through and updated all the notes to this convention, hope that works 👍

```scss
@use '@carbon/react' with ($css--default-type: true, $css--reset: true);
```

If you were using any feature flags in v10, you can safely remove them in v11.

<InlineNotification>

You can also use `@import` to bring in Carbon, if you prefer, although `@use` is
recommended.

</InlineNotification>

If you were bringing parts of Carbon, you'll need to update the paths to reflect
the new paths in `@carbon/styles`. In general, most paths moved from
`scss/globals/scss/filename` to `scss/filename`.

```scss
// Configuration
@use '@carbon/react/scss/config' with
  ($css--default-type: true, $css--reset: true);
// Reset
@use '@carbon/react/scss/reset';
// Grid
@use '@carbon/react/scss/grid';
// Helpers
@use '@carbon/react/scss/theme';
@use '@carbon/react/scss/spacing';
@use '@carbon/react/scss/type';
// Components
@use '@carbon/react/scss/components/accordion';
@use '@carbon/react/scss/components/button';
```

For a full list of the paths that have changed in `carbon-components`, check out
our
[Migration Docs](https://github.com/carbon-design-system/carbon/blob/main/docs/migration/v11.md#changing-import-paths-from-scssglobalsscss-to-scss).

> _Note: if you see any references to `@carbon/styles` for style imports, you can
> safely replace the beginning of the path with `@carbon/react` for the same
> import in this package._

Reviewer 2: Update this one too
```suggestion
**Note:** if you see any references to `@carbon/styles` for style imports, you can
safely replace the beginning of the path with `@carbon/react` for the same
import in this package.
```

Author: Done 👍

For more details about the changes to Tabs, see our storybook documentation
[here](https://react.carbondesignsystem.com/?path=/docs/components-tabs--default).

### Step 9: Done with @carbon/react!

And that's it! You're done. At this point you have migrated to use Carbon v11
using the `@carbon/react` package.

> If you run into any problems after this point, pleasefeel free to reach out to

Reviewer 2: ```suggestion
If you run into any problems after this point, please feel free to reach out to
```

that your layouts will continue to work the same as in v10.

### Step 8: Update color tokens

If you are using color tokens from Carbon, you can either update to use the new
tokens in v11 or use the compatability theme for incremental adoption of the new
tokens while maintaining existing work from v10.

For an overview of the changes to tokens, check out our
[Migration Docs](https://github.com/carbon-design-system/carbon/blob/main/docs/migration/v11.md#design-tokens).

If you would like to use the compatability theme, you can write the following in
your project where you are currently bringing in theme:

```scss
@use '@carbon/themes/scss/compat/themes' as compat;
@use '@carbon/themes/scss/themes';
@use '@carbon/themes/scss/theme' with
  ($fallback: compat.$g100, $theme: themes.$g100);
```

Doing this will allow you to use tokens from the g100 theme in v10 along with
the tokens in v11. This will work for any of the themes in v10 including white,
g10, g90, and g100.

### Step 9: Done with @carbon/styles!

And that's it! You're done. At this point you have migrated to use Carbon v11
using the `@carbon/styles` package.

> If you run into any problems after this point, pleasefeel free to reach out to

Reviewer 2: ```suggestion
If you run into any problems after this point, please feel free to reach out to
```
